pkg/ingester/instance.go

```diff
@@ -721,6 +729,10 @@ outer:
 		if chunkFilter != nil && chunkFilter.ShouldFilter(stream.labels) {
 			continue
 		}
+
+		// To Enable downstream deduplication of sharded streams, remove the shard label
+		stream.labels = labels.NewBuilder(stream.labels).Del(distributor.ShardLbName).Labels()
```

Comment:
I'm worried that we're doing this on an ancillary function right here. This is mutating the underlying stream, which is confusing (it only mutates the stream the first time it's passed over by this function) and likely causes bugs with WAL checkpoints.

I don't think we should mutate the labels for a stream, but instead just filter it out when serving queries.